src/rgw/rgw_sal_motr.h

```diff
@@ -85,25 +322,37 @@ class MotrMetaCache
   // Make the local cache entry invalid.
   void invalid(const DoutPrefixProvider *dpp, const std::string& name);
 
-  // TODO: Distribute_cache() and watch_cb() now are only place holder functions.
-  // Checkout services/svc_sys_obj_cache.h/cc for reference.
-  // These 2 functions are designed to notify or to act on cache notification.
-  // It is feasible to implement the functionality using Motr's FDMI after discussing
-  // with Hua.
-  int distribute_cache(const DoutPrefixProvider *dpp,
-                       const std::string& normal_name,
-                       ObjectCacheInfo& obj_info, int op);
-  int watch_cb(const DoutPrefixProvider *dpp,
-               uint64_t notify_id,
-               uint64_t cookie,
-               uint64_t notifier_id,
-               bufferlist& bl);
-
   void set_enabled(bool status);
 
-  MotrMetaCache(const DoutPrefixProvider *dpp, CephContext *cct) {
-    cache.set_ctx(cct);
+  // As explained above, metadata cache doesn't use dedicated
+  // indices to pass notification. Instead, it makes use of the
+  // metada index. For example, bucket index for object metadata.
+  // attach_cache_notifi() is to attach a notification a key/value pair
+  // when an object's metadata is changed. The notification contains
+  // fdmi filter string which will trigger a fdmi event. The fdmi event
+  // will be monitored by watchers of other RGW instance. Once receiving
+  // fdmi event, the callback function of the watcher is invoked to retrieve
```

Comment:
This is good explanation. I'd suggest mentioning the callback function name here also.